taichi/backends/cc/cc_program.cpp

```diff
@@ -41,7 +41,7 @@ void CCProgramImpl::materialize_snode_tree(
   layout_ = gen.compile();
   size_t root_size = layout_->compile();
   size_t gtmp_size = taichi_global_tmp_buffer_size;
-  size_t args_size = taichi_max_num_args * sizeof(uint64);
+  size_t args_size = taichi_result_buffer_entries * sizeof(uint64);
```

Comment:
Why this change?

Reply:
Because the arg buffer and the result buffer are the same, and `result_buffer_entries` is greater than `max_num_args`